### ISSUE TITLE
tests fail on i686

### ISSUE THREAD

**Issue by rekado** [haskell/tar#21]
8 tests fail when building the latest release on i686.  Here's the complete log:
https://hydra.gnu.org/build/1482909/log/raw

**Comment by clinty** [comment]
That log seems to no longer be there so I can't be sure that this is the same issue, but these tests are not 32-bit-safe:

```
    lookup:       FAIL (0.02s)
      *** Failed! Exception: 'Enum.fromEnum{Word32}: value (2172266651) is outside of Int's bounds (-2147483648,2147483647)' (after 48 tests): 
      ValidPaths [("a/a/b/a/c/a/b/b/b/d/d/b/d/b/b/b/c/d/c/a/d/d/d/d/b/b/c/c/c/c/b/d/c/d/b/d",(7186129657,0)),("a/c/a/b/d/b/b/c/c/a/a/a/a/d/d/a/a/a/d/d/b/a/c/b/b/a/c/c/b/c/c/b/a/d/b/b",(0,14035411)),("d/c/c/a/d/b/a/d/b/b/c/a/d/b/b/b/d/d/a/d/b/b/a/c/c/c/d/d/a/a/a/c/d/d/a/a/b/b/b/d/c/d",(87431524690,14035412)),("d/c/b/c/a/c/d/a/c/d/b/b/d/b/c/b/d/d/b/a/b/b/d/c/c/c/d/b/b/a/a/b/b/c/c/a/c/d/a/d/b/a/d",(32663129786,184800110)),("a/b/b/a/a/b/d/d/d/c/c/c/c/b/d",(24663837556,248595287)),("b/b/b/a/b/d/d/b/b/c/d/c/d/c/d/d/b/c/c",(100805507232,296766846)),("b/b/b/b/b/a/a/c/a/b/a/b/a/d/b/b/b/b/a/c/b/d/b/a/c/d/a/a/c/b/c/c/d/d/c/d/a/d",(0,493652604)),("a/b/c/a/d/b/a/d/a/b/c/c/b/a/c/b/d/c/a/c/b/b/a/c/d/c/d/a/a/c/b/d/d/c/a/b/d/d/b/a",(58299293221,493652605)),("b/a/b/d/b/a/a/c/c/c/c/d/b/a/b/c/a/b/b/b/b/d/d/b/c/a/d/c/d/c/c/b/c/c/d/a/a/a/d/b/d/a",(38403717432,607518414)),("b/d/b/c",(90206060209,682525676)),("a/a/d/b",(0,858709389)),("b/b/d/b/a/b/c/d/a/b/a/d/c/d/c/d/c/b/a/a/c/c/c/a/b/b/b/a/c/b/d/d/c/d/d/d/d",(114883399237,858709390)),("d/b/a/c/c/d/a/d/b",(7260420328,1083091031)),("d/a/d/d/d/a/d/b/a/d/a/b/d/c/d/a/a/d/c/b/a/d/a/a/d/b/d/d/c/d/b/a/b/c/c/c/a/b/a/c/d/d/a/a/b/d",(101986669727,1097271541)),("c/a/d/b/c/b/d/b/d/a/a/a",(136920202455,1296464257)),("b/a/a/c/b/a/c/d/b/a/d/d/b/a/a/a/d/d/b/c/c/a/b/c/d/b/d/b/a/d/c/a/a/a/d/a/b/b",(32053209358,1563886529)),("d/b/c/c/c/c/d/b/d/b/c/a/d/a/c/d/a/d/a/c/b/c/a/d/d/c/a/c/b/c/b/b/c/a",(0,1626490455)),("a/b/c/c/c/b/a/d/c/d/d/d/a/a/d/a/c/c/b/a/d/b/a/d/b/b/b/c/a/d/b/a/b/d/d/a/d",(33737608146,1626490456)),("a/d/a/d/c/d/c/c/b/a/c/c/d/b/c/d/a/c/b/c/d/d/a/d/d/a/c/b/a/c/d/d",(126053125296,1692384223)),("c/c/b/b/d/c/c/b/a/d/d/c/a/b/c/c/d/c/a/b/d/c/a/d",(0,1938581735)),("a/c/c/d/a/b/b/c/d/d/c/c/a/a/b/b/d/a/a/d/a/b/d/c/a/b/c/d/a/a/d/a/c/a/d/b",(119646675719,1938581736)),("a/d/a/a/c/c/a/c/d/c/a/d/d/a/d/d/a/c/b/b/c/c/c/b/a/c/b/a/c/a/a/c/d/d/a/d/d/c/c/d/c/d/c/d/a/b/d",(49108470992,2172266651)),("c/d/c/d/d/a/b/c/c/c/a/a/b/b/a/a/b/a/b/a/c/b/a/c/a/d/c/a/a/b",(0,2268181635)),("a/d/c/b/a/a/a/b/c/d/c/d/a/c/c/c/b/c/b/c/b/b/d/c/a/a/b/c/d/c",(105948059206,2268181636)),("c/d/b/b/b/a/d/a/c/d/c/c/b/c/a/c/d/b/c/b/d/a/a/a/a/a/a/c/b/a/b/b/a/a/b/a/c/c/a/b/c/b/a/a/d/c",(119187132668,2475111441)),("d/d/a/d",(0,2707898811))]
      NonEmptyFilePath "b/b/b"
      Use --quickcheck-replay=327069 to reproduce.
    valid:        FAIL (0.04s)
      *** Failed! Exception: 'Enum.fromEnum{Word32}: value (2206116387) is outside of Int's bounds (-2147483648,2147483647)' (after 48 tests): 
      ValidPaths [("a/c/c/a/b/d/c/d/a/a/c/c/d/a/a/a/c/c/a/b/c/c/b/b/c/d/b/b/a/b/c/d/c/c/d/c/b/c/a/c/c/d/b/b/c/a/b",(67812205794,0)),("a/d/b/c/b/d/c/b/d/b/d/a/a/a/b/a/d/c/c/c/b/a/a/b/c/d/b/c/b/b/a/b/d/b/b/b/c/b/b",(0,132445716)),("d/d/a/d/b/a/c/a/c/c/a/d/b/c/b/d/a",(54109373009,132445717)),("b/b/a/c/a/d/b/a/d/c/b/c/d/a/c/b/b/b/b/b/d/c/c/a/a/b/c/b/b",(115194648036,238128088)),("b/c/a/d/b/a/b/a/c/a/a/d",(126321768242,463117636)),("b/b/a/a/c",(0,709839841)),("a/c/c/b/d/c/b/c/b/c/d/c/a/a/d/c/d/c/d/c/c/c/c/b/d/c/c/c/d/a/d/a/a",(55961058826,709839842)),("b/b/c/c/b/c/a/a/a/c",(23055882302,819138787)),("d/b/a/c/a/a/d/a/b/a/b/a/a/c/a/d/d/c/b/a/b/c/b/b/b/d/d/b/a/d/a/c/b/d/d/d/d",(0,864169809)),("c/d/a/b/c/a/d/d/d/d/c/c/c/b/d/d/b/c/b/b/a/b/d/b/a/c/a",(35236360063,864169810)),("a/d/a/c/b/c/b/a/d/d/d/d/b/a/b/d/a/c/c/b/a/a/d/c/b/d/a/b/a/b",(117806451513,932990827)),("a/a/a/b/d",(89422203494,1163081554)),("b/c/d/a/d/b/a/a/d/a/d/d/b/a/c/a/a/d/c/c/c/b/b/d/a/c/d/a/a/d/a",(0,1337734297)),("b/a/c/b/c/d/b/c/c/c/a/b/d/b/c/c/b/b/d/d/d/b/c/c/d/a/d/b/c/a/d/a",(16385171870,1337734298)),("a/d/d/d/b/a/d/d/c/c/a/c/c/a/d/c/d/b/a/b/b/a/c/c/d/a/c/d/c/c/a/c/a",(49127755852,1369736588)),("a/a/d/d/c/d/a/b/d/a/c/c/d/b/a/a/b/c/c/d/d/a/b/d/d/a/a/b/a/b/b/c/b/b/d/b/c/c/d/d",(0,1465689238)),("b/c/d/a/b/c/b/c/b/a/b/b/c/c/b/d/d/d/a/b/b/c/b/a/a/d/a/d/d/d/b/a/d/a/c/a/c/a/a/a",(117324870334,1465689239)),("c/a/c/c/a/a/a/a/a/d/d/b/c/d/c/d/d/d/b/a/b/c/d/b/d/d/c/d/b/d/c/b/a/d/d/d/a/b/b/c/d/d",(3638965263,1694839378)),("a/d/c/b/d/a/c/c/a/d/c/c/d/a/a/d/b/b/b/c/d/b/a/a/c",(18836735499,1701946734)),("b/c/b/a/a/d/b/b/d/c/d/c/a/c/d/a/a/d/d/d/c/b/b/c/b/a",(68659135769,1738737235)),("b/b/c/b/d/d",(78777850238,1872837111)),("a/d/d/a/a/b/b/a/c/c/c/d/d/a/b/d/a/d/d/a/a/b/a/a/c/c/a/b/a/b/b/d/b/a/c/b",(0,2026700101)),("a/c/a/b/c/b/a/c/b/d/d/d/b/b/b/a/d/a/a/a/b/c/a/b/c/a/d/d/b/b/c/b/b/a/d/d/b/b/b/c/d/a/b/c/d/d",(91861137399,2026700102)),("d/c/d/d/c/a/d/a/b/c/b/d/c",(133119466318,2206116387)),("b/c/d/b/d/c/b/c/b/b/c/d/a/d/a/d/b/d/a/c/a/a/c/b/a/d/c/b/d/a/b/d/d/b/a/b/b/a/a/b/a/a",(53880957818,2466115346)),("b/a/b/d/c/a/b/b/a/b/b/b/c/b/a/b/d/d/d/c/a/b/d/c/d/d/b/c/c/c/b/d/a/d/a/b/b/a/d/d/c/d",(0,2571351593))]
      Use --quickcheck-replay=286841 to reproduce.
    toList:       FAIL (0.02s)
      *** Failed! Exception: 'Enum.fromEnum{Word32}: value (2509426566) is outside of Int's bounds (-2147483648,2147483647)' (after 49 tests): 
      ValidPaths [("a/b/d/c/b/a/b/a/d/b/b/d/c/a/d/b/c/d/c/d/c/d/b/d/b/b/a/b/c/b/b/c/d/a",(194073601931,0)),("a/d/d/d",(7035741810,379050005)),("a/a/d/b/b/b/a/b/a/d/d/a/d/b/d/d/b/c/a/d/c/b/b/c/d/b/a/d/d/b/d/c/c/a/b/c/a/c/a",(100720453850,392791690)),("d/d/a/c/d/b/c/a/c/d/d/c/b/c/d/b/b/b/b/c/a/c/d",(257616125087,589511328)),("c/a/c/a/a/d/d/c/c/b/c/c/d/c/b",(0,1092667824)),("d/a",(154441057514,1092667825)),("c/b/c/a/c/a/d/a/c/c/b/b/b/a/d/a/a/d/a/a/c/d",(0,1394310517)),("c/c/c/d/c/b/c/b/b/d/a/a/d/d/c/b/a/c/d/c/b/d/a/c/c/d/c/d/b/b/b/d/c/b/c/d/c/b/c/c/b/c",(178546070308,1394310518)),("a/b/b/d/c/a/b/a/b/a/b/c/d",(189485978901,1743033313)),("a/c/d/a/c/a/b/a/b/d/a/d/c/c/b/a/b/b/b/c",(17231318511,2113123117)),("d/b/c/a/b/b/a/a/b/d/d/b/c/c/d/c/d/c/b/a/c/a",(0,2146778037)),("a/b/d/c/b/a/a/a/a/b/d/d/c/a",(185676045380,2146778038)),("d/b/a/b",(129767121285,2509426566)),("c/a/a",(57029079297,2762877976)),("b",(0,2874262898))]
      Use --quickcheck-replay=642860 to reproduce.
    serialise:    FAIL (0.02s)
      *** Failed! Exception: 'Enum.fromEnum{Word32}: value (2349998654) is outside of Int's bounds (-2147483648,2147483647)' (after 48 tests): 
      ValidPaths [("a/b/b/a/b/d/b/b/a/d/a/b/d/d/a/c/b/d/b/a/c/a/b/b/b/a/a/a/d/c/a/b/d/d",(0,0)),("b/c/c/d/a/a/b/d/b/d/a/a/c/a/b/c/a/a/c/b/a/d/a/c/b/a/a/c/d/d/d/a/a",(56755083563,1)),("c/b/d/b",(76438006598,110849775)),("d/c/b/c/c/b/b/d/c/a/c/a/d/b/d/a/a/b/b/c/c/d/a",(2001368,260142758)),("a/d/c/c/a/c/b",(78580814356,260146668)),("d/d/c/d/c/b/c/c/b/b/d/d/d/b/b/d/c/d/c/c/d/b/c/d/c/b/c/b/a/c/b/b/a/d/c/d/d",(123787673636,413624823)),("b/d/d/d/d/c/c/a/a/a/c/b/b/d/a/b/a/d/b/b/d/c/a/a/b/a/d/d/d/d/a/c/b/d/c/a/b/b/d/b/b/d/d/a/c/a/d",(32209138152,655397625)),("d/d/d/d/b/d/d/c/d/d/c/a/a/d/c/b/d/a/d/d/b/d/b/a/b/b/c/c/c/b/a/c/b/a/b",(122853301420,718306099)),("d/d/a/a/b/a/c/b/c/c/c/b/b/c/d",(92036430054,958253955)),("d/b/d/b/c/a/b/b/d/a/b/a/b/c/a/b/c/a/b/d/a/b",(19852748582,1138012609)),("b/b/c/c/c/c/a/c/a/b/d/b/b/d/b/d/a/a/c/d/b/b/c/b/b/d/b/a/b/b/c/d/b/d/a/c/c/c/a/c/d/a/d/c/c/b/b",(98748737474,1176787510)),("d/d/b/d/d/c/c/c/a/a/c/d/d/b/c/c",(101601111656,1369656139)),("a/b/c/b/d/d/b/a/c/d/d/a/a/a/a",(36461085118,1568095812)),("b/d/d/c/c/d/c/d/b/c/b/a/d/a/b/a/a/d/c/a/c/d/c/c/a",(0,1639308870)),("b/a/a/d/c/c",(80372643257,1639308871)),("c/a/c/c/d/d/b/d/a/a/a/c/a/b/b/c/c/a/a/a/b/b/b/c/a/a",(0,1796286691)),("d/b/c/c/b/c/b/d/a/a/a/b/c/b/d/c/b/d/d/b/b/b/d/a/d/a/d/c/a/b/b/d",(41780061781,1796286692)),("b/a/c/d/d/b/b/a/c/a/b/b/c/d/a/c/c/d/b/a/d/b/d/c/d/a/b",(24900517279,1877888377)),("a/a/a/b/d/a/b/a/a/b/c/c/d/c/d/a/c/a/c/c/c/c/c/a/a/a/a/d/c/d/c/a/a/a/d/c/c/d/c/a/d/d/d",(34211437648,1926522201)),("a/b/c/d/b/a/a/c/c/b/a/d/c/c/c/d/d/b/b/b/c/d/a/d/a/c/d/b/d/d/a/d/a/a/b/c/a/b/c/a",(49578520349,1993341417)),("b/c/b/a/c/d/d/b/c/b/a/c/c/c/a/b/b/d/a/a/b/d/b/b/c/b/b/c/c/b/d/b",(133029983311,2090174466)),("c/b/c/b/b/a/a/a/a/a/b/d/a/a/a/c/b/b/c/c/c/c/d/b/b/c/a/d/b/a/d/c/a/c/d/c/a/a/d/c/b",(131182558259,2349998654)),("c/c/d/b/a/a/b/d/c/b/c/d/a/a/c/d/b/a/d/c/a/b/c/c/b/d/c/b/a/b/a/d/d/b/a/d/c/c/c/b",(35093588288,2606214590)),("c/c/d/c/c/d/c/d/d/b/d/a/d/a/d/c/c/b/c/a/d",(0,2674756756)),("b/a/a/a/d/b/d/d/c/d/c/d/a/d/d/a/b",(4517001606,2674756757)),("a/b/c/c/d/a/b",(99701040213,2683579027)),("d/a/b/c/d/d/a/b/a/d/c/c/b/a/b/d/c/b/b/d/d/b/a/a/a/b/c/b/c/d/b/a/d/c/b/a/d/d/a/c/b/b/a/c/a/c",(41203157016,2878307623)),("b/b/a/b/c/c/c/c/c/d/d/c/a/a/a/a/c/c/d/d/d/a/a/d/c/c",(132659505120,2958782541)),("a/b/a/a/c/b/a/d/d/c/b/a",(97121462165,3217883138)),("d/c/d/b/b/b/c/b/b/d/d/d/d/d/a/d/a/b/a/b/b/c/c/d",(102529646635,3407573495)),("d/d/d/d/d/c/a/c/b/b/c/b/b/b/b/b/c/d",(4578885043,3607826713)),("b/c/c/b/c/c/b/d/b/b/b/c/d/a/a/b/c/a/a/c/a/d/c",(110925644592,3616769849))]
      Use --quickcheck-replay=251204 to reproduce.
    size:         FAIL (0.02s)
      *** Failed! Exception: 'Enum.fromEnum{Word32}: value (2284973121) is outside of Int's bounds (-2147483648,2147483647)' (after 49 tests): 
      ValidPaths [("a/d/d/b/d/c/b/a/c/b/a/c/d/c/b/a/c/b/c/c/c/c/b/b/b/a/d",(0,0)),("a/c/d/c/c/c/b/a/b/c/a/b/c/a/c/c/a/c/b/a/b/d/c/b/a/b/b/d/a/b/a/a/a/d/d/c/c/c/a",(145876823970,1)),("a/d/d/d/b/b/a/a/c/a/d/d/b/a/c/d/a/d/c/b/a/d/b/a/b/d/a/a/b/c",(192574678085,284915674)),("a/c/a/c/a/a/d/a/c/a/d/d/c/d/a/c/b/d/b/a/c/a/a/b/d",(57383316955,661038094)),("a/b/b/c/b/a/d/b/d/c/a/d/c/b/b/d/a",(224756846871,773114886)),("c/a/a/a/b/a/d/d/b/c/c/a/a/a",(183923669184,1212093104)),("c/a/b/c/c/d/b/b/b/c/c/d/c/a/b/b/c/c/d/a/b/d/c/d/d/b",(0,1571319022)),("b/d/c/c/d/c/b/b/a/a/c/d/c/c/c/d/d/a/d/b/a/b/a/b/b/d/c",(151388736863,1571319023)),("a/b/c/c",(214002159850,1867000151)),("c/b/b/c/c/d/b/d/a/a/a/d/a/d/a/b/b/d/b/a/c/c/b/a/b/d/c/b/c/d/b/b/a/b/d/c/b/b/b/c/c/c/a",(162929632120,2284973121)),("c/c/a/c/a/c/d",(225856837223,2603195060)),("b/c/a/d/d/c/a/d/c/b/b",(109489600025,3044321697)),("b/a/d/c/c/d/b/d/b/b/a/a/a/b/c/c/d/d/a/a/b/c/a/a/d/b/a/d/c/b/a/b/a/d/c/a/c/d/d",(0,3258168574)),("a/d/a/a/b/d/b/d/b/a/d/a/b/c/c/a/c/a/b/d/d/d/b/c/d/a/b/d/d/d/d/c/a/a/d/d/a/b/a",(85494169855,3258168575)),("a/c/c/c/a/d/a/b/b/d/b/d/c/d/d",(256887011245,3425149377)),("b/c/b/d/c/d/c/a/a/b/b/d/a/b/c/c/c/b/d/b/d/d/d/d/a/c/c/d/a/a/a/b/a/d/b/b/d/a/c/c/d/b/a/c/d/d/a/d",(267930286785,3926881822)),("c/a/b/a/a/c/d/d/b/b/a/a/a/c/b/d/b/a/c/c/d/b/c/c/d/c/d/c/b/c/d/a/b/c/a/b",(39718455488,155215869)),("c/d/c/b/d/b/b/a/b/a/c/d/b/d/c/c/d/c/b/b/d/c/b",(0,232790979)),("b/a/a/a",(233417715287,232790980)),("c/a/b/d/c/a/c/a/a/a/c/b/d/a/b/d/b/c/b/c",(13826127312,688684957)),("c/d/a/a/c/b/d/c/b/a/c/b",(260778586962,715689113)),("c/d/c/b/a/d/c",(109357625220,1225022292)),("b/b/c/c/d/d/c/d/b/b/c/d/d/a/d/a/b/b/d/a/b/b/b/c",(170779409570,1438611405)),("b/d/b/b/b/a/c/c/b/a/c/b/a/b/a/a/c/b/b/a/a/c/b/a/b/d/b/d/c/a/a/b/a",(85386851940,1772164941)),("c/b/c/d/a/c/c/d/a/b/d/c/d/c/c/c/a/b/c/a/c/c/b/d/d/d/a/a/c/c",(67642170129,1938936138)),("b/d/d/b/d/c/d/c/b/c/c/b/d/d/a/d/b/b/d/a/d/b/a/b/a/a/d/b/b/b",(133517074331,2071049753)),("a/b/c/d",(233501151042,2331825290)),("a/b/b/d/a/b/b/c/c",(221243193179,2787882227)),("c/a/c/c/c/b/a/c/b/a/a/c/a/a/d/a/a/a/b/a/c/a/b/b/d/a/c/d/a/a",(137784789677,3219997840)),("c/c/c/d/b/c/c/b/c/a/d/b/b/d/a/a/b/c/b/b/a/a/a/d/d/a",(90921199022,3489108759)),("d/b/a/a/c/b/b/b/c/b/b",(85790332099,3666689227)),("d/d",(130373439809,3834248471)),("c/a/d/c",(105792353812,4088884097))]
      Use --quickcheck-replay=26665 to reproduce.
```

**Comment by Bodigrim** [comment]
Thanks, fixed in 18fbe84af06326469840831bb60ae0ff355f0784.
